lib/Runtime/Library/JavascriptNumber.cpp

```diff
@@ -1032,49 +1034,42 @@ namespace Js
 
         JavascriptString *dblStr = JavascriptString::FromVar(FormatDoubleToString(value, NumberUtilities::FormatFixed, -1, scriptContext));
         const char16* szValue = dblStr->GetSz();
+        const size_t szLength = dblStr->GetLength();
 
-#ifdef ENABLE_GLOBALIZATION
-        count = GetNumberFormatEx(LOCALE_NAME_USER_DEFAULT, 0, szValue, NULL, NULL, 0);
+        pszRes = szRes;
+        count = Numbers::Utility::NumberToDefaultLocaleString(szValue, szLength, pszRes, bufSize);
 
-        if( count <= 0 )
+        if ( count == 0 )
         {
             return dblStr;
         }
         else
         {
-            if( count > bufSize )
+            if ( count > bufSize )
             {
                 pszRes = pszToBeFreed = HeapNewArray(char16, count);
-            }
-            else
-            {
-                pszRes = szRes;
-            }
 
-            int newCount = GetNumberFormatEx(LOCALE_NAME_USER_DEFAULT, 0, szValue, NULL, pszRes, count);
+                count = Numbers::Utility::NumberToDefaultLocaleString(szValue, szLength, pszRes, bufSize);
```

Comment:
Should you be calling this with count instead of bufSize since count > bufSize, and we just allocated a larger array.

Also, in the case where count > bufSize, before we did a max of two calls to GetNumberFormatEx. Now it looks like we could do 4? That is, the first call to NumberToDefaultString would do it with 0, and then bufSize, and then the second would again do it with 0, and then with count. Worth optimizing to match the earlier performance?